ui/pages/smart-transactions/smart-transaction-status-page/smart-transaction-status-page.tsx

```diff
@@ -0,0 +1,314 @@
+import { hideLoadingIndication } from '../../../store/actions';
+
+type RequestState = {
+  smartTransaction?: SmartTransaction;
+  isDapp: boolean;
+};
+
+export type SmartTransactionStatusPageProps = {
+  requestState: RequestState;
+  onCloseExtension: () => void;
+  onViewActivity: () => void;
+};
+
+export const showRemainingTimeInMinAndSec = (
+  remainingTimeInSec: number,
+): string => {
+  if (!Number.isInteger(remainingTimeInSec)) {
+    return '0:00';
+  }
+  const minutes = Math.floor(remainingTimeInSec / 60);
+  const seconds = remainingTimeInSec % 60;
+  return `${minutes}:${seconds.toString().padStart(2, '0')}`;
+};
+
+export const SmartTransactionStatusPage = ({
+  requestState,
+  onCloseExtension,
+  onViewActivity,
+}: SmartTransactionStatusPageProps) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const { smartTransaction, isDapp } = requestState;
+  const isSmartTransactionPending =
+    !smartTransaction ||
+    smartTransaction.status === SmartTransactionStatuses.PENDING;
+  const isSmartTransactionSuccess =
+    smartTransaction?.status === SmartTransactionStatuses.SUCCESS;
+  const isSmartTransactionCancelled = smartTransaction?.status?.startsWith(
+    SmartTransactionStatuses.CANCELLED,
+  );
+  const featureFlags: Record<string, any> | null = useSelector(
+    getFeatureFlagsByChainId,
+  );
+  const stxEstimatedDeadline =
+    featureFlags?.smartTransactions?.expectedDeadline || 45;
+  const stxMaxDeadline = featureFlags?.smartTransactions?.maxDeadline || 150;
+  const [timeLeftForPendingStxInSec, setTimeLeftForPendingStxInSec] =
+    useState(stxEstimatedDeadline);
+  const [isSmartTransactionTakingTooLong, setIsSmartTransactionTakingTooLong] =
+    useState(false);
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider, shallowEqual);
+  const chainId: string = useSelector(getCurrentChainId);
+  const stxDeadline = isSmartTransactionTakingTooLong
+    ? stxMaxDeadline
+    : stxEstimatedDeadline;
+
+  const txHash = smartTransaction?.statusMetadata?.minedHash;
+
+  const baseNetworkUrl: string =
+    rpcPrefs.blockExplorerUrl ??
+    SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP[chainId] ??
+    '';
+
+  let blockExplorerUrl: string = '';
+  if (txHash) {
+    blockExplorerUrl = getBlockExplorerLink(
+      { hash: txHash, chainId, metamaskNetworkId: '' },
+      { blockExplorerUrl: baseNetworkUrl },
+    );
+  }
+  let title;
+  let description;
+  let iconName;
+  let iconColor;
+  let countdown;
+
+  if (isSmartTransactionPending) {
+    countdown = (
+      <Text
+        display={Display.InlineBlock}
+        textAlign={TextAlign.Center}
+        color={TextColor.textAlternative}
+        variant={TextVariant.bodySm}
+        className="smart-transaction-status-page__countdown"
+      >
+        {showRemainingTimeInMinAndSec(timeLeftForPendingStxInSec)}
+      </Text>
+    );
+  }
+
+  if (isSmartTransactionPending && isSmartTransactionTakingTooLong) {
```

Comment:
Could this logic be moved to a separate function such as `getDisplayValues` to help make this block smaller?

It would also remove the need for the `let` variables we could just early return for each condition.